New DD Agent probabilistic sampling option (#23481)
DataDog/documentation: content/en/opentelemetry/ingestion_sampling_with_opentelemetry.md

### Probabilistic sampling

When using Datadog Agent OTLP ingest, a probabilistic sampler is available starting with Agent 7.54.0. 

#### Configuring

> To configure probabilistic sampling, either:

Reviewer: ```suggestion
To configure probabilistic sampling, do one of the following:
```

> - Set `DD_APM_PROBABILISTIC_SAMPLER_ENABLED` to `true` and set `DD_APM_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE` to the percentage of traces you'd like to sample (between `0` and `100`)

Reviewer: ```suggestion
- Set `DD_APM_PROBABILISTIC_SAMPLER_ENABLED` to `true` and `DD_APM_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE` to the percentage of traces you'd like to sample (between `0` and `100`).
```

> - Or add the following YAML to your Agent's configuration file:

Reviewer: ```suggestion
- Add the following YAML to your Agent's configuration file:
```

  ```yaml
  apm_config:
    # ...
    probabilistic_sampler:
        enabled: true
        sampling_percentage: 50 #In this example, 50% of traces are captured.
  ```

> If you use a mixed setup of Datadog tracing libraries and OTel SDKs, probabilistic sampling will apply both to spans originated from Datadog and OTel tracing libraries.

Reviewer: ```suggestion
If you use a mixed setup of Datadog tracing libraries and OTel SDKs, probabilistic sampling will apply to spans originating from both Datadog and OTel tracing libraries.
```

> <div class="alert alert-warning"><code>DD_OTLP_CONFIG_TRACES_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE</code> is deprecated and replaced by <code>DD_APM_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE</code>.</div>

Reviewer: ```suggestion
<div class="alert alert-warning"><code>DD_OTLP_CONFIG_TRACES_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE</code> is deprecated and has been replaced by <code>DD_APM_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE</code>.</div>
```

#### Considerations

> **Notes:** 

Reviewer: ```suggestion
```

> - Probabilistic sampling is **incompatible with [head-based sampling][16]**. The probabilistic sampler with ignore spans' sampling priority set at the tracing library level. As a result, head-based sampled traces might still be dropped by probabilistic sampling. 

Reviewer: ```suggestion
- The probabilistic sampler will ignore the sampling priority of spans that are set at the tracing library level. As a result, probabilistic sampling is **incompatible with [head-based sampling][16]**. This means that head-based sampled traces might still be dropped by probabilistic sampling.
```

[13]: https://app.datadoghq.com/dash/integration/apm_estimated_usage
[14]: /opentelemetry/guide/migration/
[15]: /opentelemetry/interoperability/otlp_ingest_in_the_agent/?tab=host

> [16]: /opentelemetry/collector_exporter/

Reviewer: ```suggestion
```
